frontend/src/Context/PublisherProvider/usePublisherOptions/usePublisherOptions.tsx

```diff
@@ -7,49 +7,57 @@ import {
 } from '@vonage/client-sdk-video';
 import useUserContext from '../../../hooks/useUserContext';
 import getInitials from '../../../utils/getInitials';
+import DeviceManager from '../../../utils/DeviceManager';
 
 /**
  * React hook to get PublisherProperties combining default options and options set in UserContext
- * @returns {PublisherProperties} publisher properties object
+ * @returns {PublisherProperties | null} publisher properties object
  */
-const usePublisherOptions = (): PublisherProperties => {
+
+const usePublisherOptions = (): PublisherProperties | null => {
   const { user } = useUserContext();
+  const [publisherOptions, setPublisherOptions] = useState<PublisherProperties | null>(null);
+  const deviceManagerRef = useRef<DeviceManager>();
+
+  useEffect(() => {
+    const setOptions = async () => {
+      if (!deviceManagerRef.current) {
+        deviceManagerRef.current = new DeviceManager();
+        await deviceManagerRef.current.init();
+      }
+
+      const videoSource = deviceManagerRef.current.getConnectedDeviceId('videoinput');
+      const audioSource = deviceManagerRef.current.getConnectedDeviceId('audioinput');
+
+      const { name, noiseSuppression, blur, publishAudio, publishVideo } = user.defaultSettings;
```

Comment:
IMO, it's odd that we have these settings in `UserContext` but we don't have the other settings above in it.  Instead, they're in `DeviceManager`.

If the media sources don't belong in `UserContext`, then maybe we can add them in `AppContext`, `SystemContext`, `ConfigContext` or something.

Reply:
would that refactor be a blocker for you or could I do in a follow up PR?